cli/compose/convert/service.go

```diff
@@ -122,6 +122,11 @@ func Service(
 		}
 	}
 
+	isolation := container.Isolation(service.Isolation)
+	if !isolation.IsDefault() && !isolation.IsHyperV() && !isolation.IsProcess() {
+		return swarm.ServiceSpec{}, fmt.Errorf("Unsupported isolation %s. Valid values are %s, %s, %s", isolation, container.IsolationDefault, container.IsolationProcess, container.IsolationHyperV)
```

Comment:
This is already validated by the `isolationOpt` we shouldn't need to do it again here, right?

The validation should be handled by the flags/value types so that any value stored in the opt is valid to send to the API.

Reply:
In this case, the source is not an isolationOpt flag, but a compose-file field (which is of type string)

Reply:
Did not see that, thank! I changed validation by typing the isolation field in service type to container.Isolation, and introduced a transformValidation function at loading time (that does the validation). Also added a parsing test for invalid isolation.